src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -2850,4 +2850,129 @@ public class RequiredMemberAttribute : Attribute
             Diagnostic(ErrorCode.ERR_ExplicitRequiredMember, "RequiredMember").WithLocation(6, 6)
         );
     }
+
+    [Fact]
+    public void RequiredMemberSuppressesNullabilityWarnings_01()
+    {
+        var code = @"
+#pragma warning disable CS0649 // Field is never assigned to
+#nullable enable
+class C
+{
+    public required string P1 { get; set; }
+    public required string F1;
+    public string P2 { get; set; }
+    public string F2;
+}
+";
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics(
+            // (8,19): warning CS8618: Non-nullable property 'P2' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+            //     public string P2 { get; set; }
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "P2").WithArguments("property", "P2").WithLocation(8, 19),
+            // (9,19): warning CS8618: Non-nullable field 'F2' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.
+            //     public string F2;
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "F2").WithArguments("field", "F2").WithLocation(9, 19)
+        );
+    }
+
+    [Fact]
+    public void RequiredMemberSuppressesNullabilityWarnings_02()
+    {
+        var code = @"
+#pragma warning disable CS0649 // Field is never assigned to
+#nullable enable
+class C
+{
+    public required string P1 { get; set; }
+    public required string F1;
+    public string P2 { get; set; }
+    public string F2;
+
+    public C()
+    {
+    }
+
+    public C(int _) {}
```

Comment:
Consider testing a constructor with `[SetsRequiredMembers]`.

Reply:
This isn't implemented yet. That's next PR. However, `SetsRequiredMembers` will not affect behavior here: LDM made the decision that `SetsRequiredMembers` does not ensure that required members are set in the constructor, so no nullable warnings should be given either.

Reply:
That's fine. Let's just add the test at the appropriate time to ensure the expected behavior is recorded.